### PR TITLE
Adjust to core changes (addon.xml)

### PR DESCRIPTION
**Pull request by J-N-K**
Related to https://github.com/openhab/openhab-core/pull/3050

### CODE REVIEW

**Comment by J-N-K** [comment]
I think now that we have addon.xml for every add-on it would be good to have a check if that is present. Unfortunately I did not find out how to declare exceptions based on the module name, which is necessary e.g. for `org.openhab.mqtt.homie` (because the XML is on the "base project" `org.openhab.binding.mqtt`

File: custom-checks/checkstyle/src/test/java/org/openhab/tools/analysis/checkstyle/test/OhInfXmlValidationCheckTest.java [hunk 1]
```diff
@@ -162,7 +162,7 @@ public void testInvalidBinding() throws Exception {
 
         int lineNumber = 7;
         String[] expectedMessages = generateExpectedMessages(lineNumber,
-                "The content of element binding:binding is not complete. One of {name} is expected.");
+                "The content of element addon:addon is not complete. One of {type} is expected.");
         verifyWithPath("invalidBinding", RELATIVE_PATH_TO_BINDING, expectedMessages);
```
Comment (wborn) [inline custom-checks/checkstyle/src/test/java/org/openhab/tools/analysis/checkstyle/test/OhInfXmlValidationCheckTest.java:166]: How about also renaming the test to `testInvalidAddon` and:

```suggestion
        verifyWithPath("invalidAddon", RELATIVE_PATH_TO_ADDON, expectedMessages);
```

**Review by wborn — COMMENTED** [review]
Thanks for the updates!
There could be a small improvement:

File: custom-checks/checkstyle/src/test/java/org/openhab/tools/analysis/checkstyle/test/OhInfXmlValidationCheckTest.java (hunk 1 above)
Comment (J-N-K) [reply custom-checks/checkstyle/src/test/java/org/openhab/tools/analysis/checkstyle/test/OhInfXmlValidationCheckTest.java]: Done

**Review by wborn — APPROVED** [review]
:+1:

**Pull request merged**